### PR TITLE
[MM-23727] Make channel validation consistent on the server

### PR DESCRIPTION
**Pull request by fmunshi**
#### Summary
- ~Channel creation enforces that the channel URL must not contain `__` no matter which characters surround it so this PR changes the update and rename logic to match the create channel validations~

- This PR now removes the strict unnecessary validation (must not contain `__`) at the server level and ensures that it is consistent no matter where in the server a rename is done from.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-23727

#### QA Test steps
- Attempt to rename a channel using the rename channel modal (or the CLI) and set the url to something like `channel__name`
- Ensure that it does not save

### CODE REVIEW

**Comment by fm2munsh** [comment]
@agnivade I agree that the validation is unnecessarily strict in this case - maybe this PR should keep the validation to only `userId__userId` and instead modify the `CreateChannel` method to also validate the same way.

The main issue that I noticed was that the server was checking for `__` on create and then checking `userID__user_ID` on rename so the validation was inconsistent as seen here: https://github.com/mattermost/mattermost-server/blob/master/app/channel.go#L147

And I will talk about the webapp validation with a PM and see what their thoughts on it are.

**Comment by agnivade** [comment]
Yep, validation is inconsistent. See my comment here: https://github.com/mattermost/mattermost-server/pull/13441#issuecomment-568152388. The translations would also need to be changed. Or perhaps add a new error code with the same message in English. I should have thought of that in that PR itself :P

Yes, it would be great if we could make both webapp and server have the same validation code. And also make it consistent for createChannel and renameChannel.

Thanks !

**Comment by jespino** [comment]
Yes, my feeling here is creating or updating channels should follow the same validation, what validation I think not allowing `__` as part of the name is a bit restrictive and is not necessary (I think). Can we make the creation validation less restrictive (checking the ids too)?

File: i18n/en.json [hunk 1]
```diff
@@ -255,10 +255,6 @@
     "id": "api.channel.create_channel.direct_channel.app_error",
     "translation": "Must use createDirectChannel API service for direct message channel creation."
   },
-  {
-    "id": "api.channel.create_channel.invalid_character.app_error",
```
Comment (agnivade) [inline i18n/en.json:259]: Please also remove the translations in other files.

**Comment by agnivade** [comment]
PM review for the error message.

File: i18n/en.json (hunk 1 above)
Comment (fmunshi) [reply i18n/en.json:259]: My assumption was that we only modify `en` translations and all other translations are updated in the community translation PRs as per @jespino 's comment here https://community.mattermost.com/core/pl/jt8sdya5pjfh7bgbr5389gs9ac

Comment (agnivade) [reply i18n/en.json:259]: Hmm .. I am not really aware of this "other process". My main concern was that we are deleting a main translation. So there's not much use of keeping the other translation strings. They are just lying there. If there's another "process" which takes care of deleting those, then I'm fine.

Comment (jespino) [reply i18n/en.json:259]: Yes, exactly, the translations are added and removed only to the `en.json` file, the rest of the files are updated through our i18n process.

File: app/channel.go [hunk 2]
```diff
@@ -464,14 +460,6 @@ func (a *App) GetGroupChannel(userIds []string) (*model.Channel, *model.AppError
 
 // UpdateChannel updates a given channel by its Id. It also publishes the CHANNEL_UPDATED event.
 func (a *App) UpdateChannel(channel *model.Channel) (*model.Channel, *model.AppError) {
-	userIds := strings.Split(channel.Name, "__")
```
Comment (jespino) [inline app/channel.go:467]: I would expect to see the `IsValid` here. This function is called from multiple places and is hard to track what are not validating if we remove it from here.

File: app/channel.go [hunk 3]
```diff
@@ -144,10 +144,6 @@ func (a *App) CreateChannelWithUser(channel *model.Channel, userId string) (*mod
 		return nil, model.NewAppError("CreateChannelWithUser", "api.channel.create_channel.direct_channel.app_error", nil, "", http.StatusBadRequest)
 	}
 
-	if strings.Index(channel.Name, "__") > 0 {
```
Comment (jespino) [inline app/channel.go:147]: Same here, I would expect to have an `IsValid` here, or may one in the `CreateChannel` method, this way we have the creation and update properly verified.

File: app/channel.go (hunk 2 above)
Comment (fmunshi) [reply app/channel.go:467]: Does it make sense to add it here when the store validates it anyway and the error is propagated up?

Comment (jespino) [reply app/channel.go:467]: Yes, that is a good point, I didn't realize that. Yes, if there is already validated at the store works for me. :)

File: i18n/en.json [hunk 4]
```diff
@@ -4502,6 +4494,10 @@
     "id": "model.channel.is_valid.id.app_error",
     "translation": "Invalid Id."
   },
+  {
+    "id": "model.channel.is_valid.name.app_error",
+    "translation": "Invalid name. User ids are not permitted in channel name for non-direct message channels."
```
Comment (wiersgallak) [inline i18n/en.json:4499]: Recommend using what we had prior (includes the word "channel" before "name"):  "Invalid channel name. User ids are not permitted in channel name for non-direct message channels."

Comment (fmunshi) [reply i18n/en.json]: @wiersgallak I have updated the wording here now

**Pull request merged**